### PR TITLE
Add verifier name suffix for the duplicated template names

### PR DESCRIPTION
**Pull request by Hrom131**
- [ ] Since this PR suggests a **bug fix**, the relevant tests have been added.
- [x] Since this PR introduces a **new feature**, the update has been discussed in an Issue or with the team.
- [ ] This PR is just a **minor change**, like a typo fix.

---

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -270,27 +270,50 @@ const generateVerifiers: ActionType<GenerateVerifiersTaskConfig> = async (
...
-          )
-        ).createVerifier(verifiersType);
+        let verifierNameSuffix: string = "";
+
+        if (templateNamesCount[circuitArtifactInfo.circuitArtifact.circuitTemplateName] > 1) {
+          Object.values(circuitArtifactInfo.circuitArtifact.baseCircuitInfo.parameters).forEach(
+            (param, index, values) => {
+              if (Array.isArray(param)) {
+                throw new HardhatZKitError("Arrays in main component parameters not supported");
+              }
```
Comment (Arvolear) [inline src/index.ts:299]: Can't we linearize it or something? Throwing an error is really bad

File: src/core/dependencies/parser/CircomFilesParser.ts [hunk 2]
```diff
@@ -99,7 +101,7 @@ export class CircomFilesParser {
     const circomTemplateInputsVisitor = new CircomTemplateInputsVisitor(
       circomResolvedFile.absolutePath,
       circomResolvedFile.fileData.parsedFileData.templates[templateName].context,
-      parameterValues,
+      cloneDeep(parameterValues),
```
Comment (Arvolear) [inline src/core/dependencies/parser/CircomFilesParser.ts:104]: Why do we need to copy?

File: README.md [hunk 3]
```diff
@@ -257,5 +257,6 @@ The optional `provingSystem` parameter should only be specified if multiple prov
 ## Known limitations
 
 - Currently there is minimal support for `var` Circom variables. Some circuits may not work if you are using complex `var`-dependent expressions.
+- Currentrly there is no support for arrays as parameters of the main component.
```
Comment (Arvolear) [inline README.md:260]:
```suggestion
- Also, passing arrays as parameters of a circuit `main` component is currently not supported.
```

File: src/core/dependencies/parser/CircomFilesParser.ts (hunk 2 above)
Comment (Hrom131) [reply src/core/dependencies/parser/CircomFilesParser.ts:104]: Inside the `CircomTemplateInputsVisitor`, information about the various var variables that are inside the template is written to this Record. Without copying this information will be saved and the parameters field will contain not only parameters of the main component, but also internal information

Comment (Arvolear) [reply src/core/dependencies/parser/CircomFilesParser.ts:104]: But doesn't `cloneDeep` do a full deep copy of all the records, including the nested records?

**Review by Arvolear — APPROVED** [review]
LGTM!

**Pull request merged**